chain/pruned_block_dispatcher.go

```diff
@@ -614,6 +631,8 @@ func (d *PrunedBlockDispatcher) handleResp(req, resp wire.Message,
 			Finished:   false,
 		}
 	}
+	copyblockChans := make([]chan *wire.MsgBlock, len(blockChans))
```

Comment:
I think we must deepcopy here, because its a slice, otherwise we might run into a race below ?

Reply:
yeah I think so since `blockChans` is just a pointer to the slice keyed using `blockHash` in map `blocksQueried`.